src/fragments/fragments.ts

```diff
@@ -30,21 +30,24 @@ export const errorMessageHandling = (fragments: VerificationFragment[]): string[
     // if the error is because cannot connect to Ethereum, only return this one
     return [TYPES.SERVER_ERROR];
   }
+  if (!issuedValid && !hashValid && !identityValid) {
+    // this error is caused when the document is invalid, only keep this one
+    return [TYPES.INVALID];
+  }
```

Comment:
Notes: Moved this part from nested `if` clause inside `!issuedValid` check to its own clause outside.✅